### PR TITLE
Fix bottom tab highlight delay

### PR DESCRIPTION
**Pull request by Zakpak0**
### Details

Modified the useActiveRoute hook to return a match for all bottom tab routes.
Renamed it useActiveBottomTabRoute
Modified files to use this hook instead of previous logic when information on the currently selected bottom tab is needed.

### Fixed Issues
$ https://github.com/Expensify/App/issues/44587
PROPOSAL:https://github.com/Expensify/App/issues/44587#issuecomment-2201827887

### Tests

1. Click on the Account bottom tab to select that route
2. Verify that the Account bottom tab has a green border around it.
3. Navigate to another tab, verify that tab is highlighted and the Account tab is not.

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Click on the Account bottom tab to select that route
2. Verify that the Account bottom tab has a green border around it.
3. Navigate to another tab, verify that tab is highlighted and the Account tab is not.

### QA Steps

1. Click on the Account bottom tab to select that route
2. Verify that the Account bottom tab has a green border around it.
3. Navigate to another tab, verify that tab is highlighted and the Account tab is not.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/72584987/1e411ed4-b1cb-4e7d-8295-b8a446904c9e
</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/72584987/da557c52-180b-48d4-a5fd-be470dc431bd

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/72584987/a24e64e8-627c-41e8-b550-972583942923
</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/72584987/9aa871ac-f7aa-4f01-905f-4c6939b4a3a8

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/72584987/5bcba51b-e84a-4606-8a9c-dbf3de4ec0df
</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/72584987/8c138eb0-1565-4819-93c1-1c1c454d826d
</details>

### CODE REVIEW

**Comment by Zakpak0** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Zakpak0** [comment]
recheck

**Comment by ikevin127** [comment]
@Zakpak0 Take you time, go through the checklist and once you ticked all the boxes and added videos testing the fix (given the **Tests** steps) for all platforms, proving that the PR fixes the issue -> then I will be able to start working on the reviewer checklist and move this PR a step closer to merge.

**Note:** For checkboxes where the PR code changes don't reach, simply tick them off without any actions. Example of checkbox given your PR: `I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)`, use your best judgement to decide when this is the case based on your PRs changes.

**Comment by Zakpak0** [comment]
@ikevin127 Thanks for the advice!
I reviewed my PR and updated it appropriately.
I also fixed my merge conflict.
If there's anything else/any advice you might have I'm all ears!  ✅

**Comment by ikevin127** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by ikevin127 — APPROVED** [review]
LGTM and tests well 🚀

@luacmartins Can you please double check the code to make sure it aligns with our  style guide ? Nothing jumps out for me, but since I'm not a very experienced code reviewer - I usually rely on CME to do a final review of the code.

**Comment by luacmartins** [comment]
We have conflicts. @adamgrzybowski would you mind reviewing this PR?

**Comment by ikevin127** [comment]
@Zakpak0 Can you please sync with main and solve the conflicts when you get a chance ? Additionally, please edit the PRs title to better reflect the issue that this is fixing since `Zakpak0/bugfix issue #44587` is not specific enough.

**Comment by Zakpak0** [comment]
Hi @ikevin127 I renamed the pull request to "Fix bottom tab highlight delay" following the style of one of @luacmartins PRs.
Does that work?
Additionally, I merged main into my branch and resolved any conflicts.

**Comment by ikevin127** [comment]
@Zakpak0 Thanks! We're awaiting a final review from @adamgrzybowski to sign-off on the code changes after which, if no changes are required / conflicts arise, the PR should get merged.

**Comment by adamgrzybowski** [comment]
@WojtekBoman as you may have more context on the topic of `useActiveRoute`

**Comment by ikevin127** [comment]
> [!note]
> Some [context](https://github.com/Expensify/App/issues/44587#issuecomment-2204845088) from proposal review regarding `useActiveRoute` not working in narrow layout devices -> leading to the changes in this PR.

**Review by luacmartins — COMMENTED** [review]
TS check is failing. Changes LGTM, but I'll let @WojtekBoman review this as well, since he has more context on this navigation logic.

**Comment by ikevin127** [comment]
@Zakpak0 To fix the TS issues, you can run `npm run typecheck` while on the branch locally, once the command runs without errors you can push the fix.

**Tip**: If you move code around, it's good practice to run `npm run prettier` as well, for formatting (otherwise lint might fail).

**Comment by Zakpak0** [comment]
@luacmartins  Fixed the type issues
Typescript didn't like the fact that search results could possibly be undefined. Which makes sense, because it doesn't seem like there are any handlers for that down the component tree.
So I added a guard to show the empty state if **searchReults** are undefined which **OnyxEntry\<T>** defines as possible.
[src/components/Search/index.tsx](https://github.com/Expensify/App/compare/main...Zakpak0:Expensify-App-Fork:zakpak0/bugfix-issue-%2344587#diff-45b3b78b48b630c32789e00793e0c947cc46c853d057e945f0ecdd0287ebb750L102-L123)
```typescript
    if (isLoadingItems) {
        return (
            <>
                <SearchPageHeader
                    query={query}
                    hash={hash}
                />
                <SearchRowSkeleton shouldAnimate />
            </>
        );
    }

    const shouldShowEmptyState = searchResults && SearchUtils.isSearchResultsEmpty(searchResults);

    if (shouldShowEmptyState || !searchResults) {
        return (
            <>
                <SearchPageHeader
                    query={query}
                    hash={hash}
                />
                <EmptySearchView />
            </>
        );
    }
```

@ikevin127 Thanks, formatted the code and run the tscheck!

**Comment by luacmartins** [comment]
@ikevin127 could you please review again given the recent changes? cc @WojtekBoman as well since you're familiar with this code

**Review by luacmartins — COMMENTED** [review]
@Zakpak0 code LGTM. We have conflicts again.

**Comment by Zakpak0** [comment]
@luacmartins got it.
Synced with main
Linted, formatted, type checked and tested app.

**Review by luacmartins — APPROVED** [review]

File: src/libs/Navigation/types.ts
Comment (adamgrzybowski) [inline src/libs/Navigation/types.ts:87]: Hey @Zakpak0, I am working on some navigation changes and I noticed these new types. Could you please explain to me why this type is necessary? I don't see any navigator named `SearchNavigator`.

Comment (Zakpak0) [reply src/libs/Navigation/types.ts:87]: Np @adamgrzybowski , happy to.
If I recall correctly, I was having a hard time finding an existing type that defined the search tab's routes in isolation.
I created that to supplement the creation of this type [here](https://github.com/Expensify/App/pull/44931/files/706148d46850cdd2cbcbe8201ef32f91eebbbf04#diff-5b0e04a21dcf344490f93813e7d9c49413bbf31dba01fcd2df358cd334970806R1119-R1120).
Maybe a better name for it would be **SearchTabRoutes**.
This is my first time touching the code base and I was attempting to follow the patterns I saw.
Even if they were a bit naive.

Comment (ikevin127) [reply src/libs/Navigation/types.ts:87]: > I am working on some navigation changes

Please feel free to mini-refactor if you're working on this at the moment and you see fit, as long as this PRs changes won't throw type errors I think you're good to go!

**Pull request merged**
